pkg/monitor/lease.go

```diff
@@ -86,16 +110,139 @@ func leaseVIP(cfgPath, masterDevice, name string) error {
 		return err
 	}
 
+	lease_file := getLeaseFile(cfgPath, name)
+	if f, err := os.Create(lease_file); err != nil {
+		log.WithFields(logrus.Fields{
+			"name": lease_file,
+		}).WithError(err).Error("Failed to create lease file")
+		return err
+	} else {
+		f.Close()
+	}
+
+	if err := WatchLeaseFile(log, lease_file, iface.Name, ip); err != nil {
+		log.WithFields(logrus.Fields{
+			"filename":      lease_file,
+			"expectedIface": iface.Name,
+			"expectedIp":    ip,
+		}).WithError(err).Error("Failed to watch the lease file")
+		return err
+	}
+
 	cmd := exec.Command("dhclient", "-d", "--no-pid", "-sf", "/bin/true",
-		"-lf", getLeaseFile(cfgPath, name), "-v", iface.Name, "-H", name)
+		"-lf", lease_file, "-v", iface.Name, "-H", name)
 	cmd.Stderr = os.Stderr
 
 	return cmd.Start()
 }
 
-func leaseInterface(masterDevice string, name string) (*net.Interface, error) {
-	mac := GenerateMac(MacPrefixQumranet[:], name)
+func WatchLeaseFile(log logrus.FieldLogger, fileName, expectedIface, expectedIp string) error {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.WithError(err).Error("Failed to add a create a new watcher")
+		return err
+	}
+
+	err = watcher.Add(fileName)
+	if err != nil {
+		log.WithFields(logrus.Fields{
+			"filename": fileName,
+		}).WithError(err).Error("Failed to add a watcher to lease file")
+		return err
+	}
 
+	go func(log logrus.FieldLogger) {
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return
+				}
+
+				if event.Op&fsnotify.Write == fsnotify.Write {
+					if iface, ip, err := GetLastLeaseFromFile(log, fileName); err != nil {
+						log.WithFields(logrus.Fields{
+							"filename": fileName,
+						}).WithError(err).Error("Failed to get lease information from leasing file")
+					} else if iface != expectedIface || ip != expectedIp {
+						log.WithFields(logrus.Fields{
+							"filename":      fileName,
+							"iface":         iface,
+							"expectedIface": expectedIface,
+							"ip":            ip,
+							"expectedIp":    expectedIp,
+						}).WithError(err).Error("A new lease has been written to the lease file with wrong data")
```

Comment:
Maybe we should add the new ip in the log message?

Reply:
It is part of the fields.